File: PHRPReader/clsScanStatsExInfo.vb
Option Strict On

Public Class clsScanStatsExInfo

    Private ReadOnly mScanNumber As Integer

    ' ReSharper disable once ConvertToVbAutoProperty
    Public ReadOnly Property ScanNumber As Integer
        Get
            Return mScanNumber
        End Get
    End Property

    Public Property IonInjectionTime As Double

    Public Property ScanEvent As Integer

    Public Property MasterIndex As Integer

    Public Property ElapsedScanTime As Double

    Public Property ChargeState As Integer

    Public Property MonoisotopicMZ As Double

    Public Property MS2IsolationWidth As Double

    Public Property FTAnalyzerSettings As String

    Public Property FTAnalyzerMessage As String

    Public Property FTResolution As Double

    Public Property ConversionParameterB As Double

    Public Property ConversionParameterC As Double

    Public Property ConversionParameterD As Double

    Public Property ConversionParameterE As Double

    Public Property CollisionMode As String

    Public Property ScanFilterText As String

    Public Property SourceVoltage As Double

    Public Property Source_Current As Double

    Public Sub New(intScanNumber As Integer)
        mScanNumber = intScanNumber
    End Sub
End Class

File: PHRPReader/clsScanStatsExInfo.cs
namespace PHRPReader
{
    public class clsScanStatsExInfo
    {
        private readonly int mScanNumber;

        // ReSharper disable once ConvertToAutoProperty
        public int ScanNumber
        {
            get { return mScanNumber; }
        }

        public double IonInjectionTime { get; set; }

        public int ScanEvent { get; set; }

        public int MasterIndex { get; set; }

        public double ElapsedScanTime { get; set; }

        public int ChargeState { get; set; }

        public double MonoisotopicMZ { get; set; }

        public double MS2IsolationWidth { get; set; }

        public string FTAnalyzerSettings { get; set; }

        public string FTAnalyzerMessage { get; set; }

        public double FTResolution { get; set; }

        public double ConversionParameterB { get; set; }

        public double ConversionParameterC { get; set; }

        public double ConversionParameterD { get; set; }

        public double ConversionParameterE { get; set; }

        public string CollisionMode { get; set; }

        public string ScanFilterText { get; set; }

        public double SourceVoltage { get; set; }

        public double Source_Current { get; set; }

        public clsScanStatsExInfo(int intScanNumber)
        {
            mScanNumber = intScanNumber;
        }
    }
}
